deploy/addon/pulsar.md

```diff
@@ -0,0 +1,268 @@
+
+A topic is defined this way:
+
+`{persistent|non-persistent}://tenant/namespace/topic`
+
+Tenants and namespaces allow for grouping and subgrouping of topics.
+
+A Clever Cloud Pulsar add-on is basically a `tenant/namespace` pair that allows you to create and use topics following this pattern:
+
+`{persistent|non-persistent}://<CLEVER_CLOUD_TENANT_ID>/<ADDON_ID>/<TOPIC_NAME>`
+
+## Authorization
+
+Pulsar add-on uses (Biscuit for Pulsar)[https://github.com/CleverCloud/biscuit-pulsar] implementation which is directly pluggable to the Pulsar authentication and authorization system. Each add-on exposes its own Biscuit token.
+
+### Usage
+
+We advise you to use (`pulsarctl`)[https://github.com/streamnative/pulsarctl] provided by StreamNative. Here is an example to list topics in your add-on (in your namespace):
+
+```bash
+pulsarctl --admin-service-url https://<PULSAR_REST_URL> --auth-params <BISCUIT> --auth-plugin org.apache.pulsar.client.impl.auth.AuthenticationToken topics list <tenant>/<namespace>
+```
+
+As Biscuit is a token, you can use `AuthenticationToken("<BISCUIT>")` provided by clients librairies to authenticate to our clusters without any tweak.
+
+* C++ client
+* C# client
+* Go client
+* Java client
+* Node.js client
+* Python client
+* Rust client
+* WebSocket client
+
+### Operations
+
+The pulsar addon given Biscuit token enables you to run several operations on namespace, its policies and the related topics.
+
+#### Namespace
+
+Namespace operations authorized:
+
+```
+CREATE_TOPIC
+GET_TOPIC
+GET_TOPICS
+DELETE_TOPIC
+ADD_BUNDLE
+DELETE_BUNDLE
+GET_BUNDLE
+CLEAR_BACKLOG
+UNSUBSCRIBE
+```
+
+Namespace policies operations authorized:
+
+```
+ALL_READ
+ANTY_AFFINITY_READ
+BACKLOG_READ
+BACKLOG_WRITE
+COMPACTION_READ
+COMPACTION_WRITE
+DELAYED_DELIVERY_READ
+DELAYED_DELIVERY_WRITE
+DEDUPLICATION_READ
+DEDUPLICATION_WRITE
+MAX_CONSUMERS_READ
+MAX_CONSUMERS_WRITE
+MAX_PRODUCERS_READ
+MAX_PRODUCERS_WRITE
+MAX_UNACKED_READ
+MAX_UNACKED_WRITE
+OFFLOAD_READ
+PERSISTENCE_READ
+PERSISTENCE_WRITE
+RATE_WRITE
+RATE_READ
+RETENTION_READ
+RETENTION_WRITE
+REPLICATION_READ
+REPLICATION_RATE_READ
+SCHEMA_COMPATIBILITY_STRATEGY_READ
+SCHEMA_COMPATIBILITY_STRATEGY_WRITE
+SUBSCRIPTION_AUTH_MODE_READ
+SUBSCRIPTION_AUTH_MODE_WRITE
+ENCRYPTION_READ
+ENCRYPTION_WRITE
+TTL_READ
+TTL_WRITE
+```
+
+#### Topics
+
+Topics operations authorized:
+
+```
+LOOKUP
+PRODUCE
+CONSUME
+COMPACT
+EXPIRE_MESSAGES
+OFFLOAD
+PEEK_MESSAGES
+RESET_CURSOR
+SKIP
+TERMINATE
+SUBSCRIBE
+GET_SUBSCRIPTIONS
+UNSUBSCRIBE
+GET_STATS
+```
+
+Topics policies operations authorized:
+
+```
+PARTITION_READ
+PARTITION_WRITE
+```
+
+### Attenuation
+
+The Pulsar add-on given Biscuit token can be attenuated, here is an attenuation example using (biscuit-cli)[https://github.com/biscuit-auth/biscuit-cli] from the given Biscuit token to produce/consume topics starting with `"TOPIC_PREFIX"`:
```

Comment:
Could you add an attenuation example?

For example, "I want to derive a biscuit from the one I have that can only read data"

Reply:
I think the attenuation example is already provided.

Reply:
It is.